airflow/cli/commands/connection_command.py

```diff
@@ -133,6 +134,24 @@ def _valid_uri(uri: str) -> bool:
     return uri_parts.scheme != '' and uri_parts.netloc != ''
 
 
+def _valid_conn_type(conn_type: str) -> Tuple[bool, List[str]]:
```

Comment:
Eagerly accessing valid conn types looks wrong to me (it’s slow). The `_get_connection_types()` function should live outside of this, and only called when `conn_type` is not valid.

This function should simply return a bool to indicate whether the conn type is valid.